src/components/shapes/attributes.js

```diff
@@ -109,6 +109,10 @@ module.exports = templatedArray('shape', {
         description: [
             'Sets the annotation\'s y coordinate axis.',
             axisPlaceableObjs.axisRefDescription('y', 'bottom', 'top'),
+            'If the axis `type` is *log*, then you must take the',
+            'log of your desired range.',
+            'If the axis `type` is *date*, then you must convert',
+            'the date to unix time in milliseconds.'
```

Comment:
Oh hmm, for shapes actually I don't think either of these is true, despite `xref` containing this text for 6 years! See `shapes.json`:

https://github.com/plotly/plotly.js/blob/b194d8c3ef503ac1989666547cccb65dc60fb44f/test/image/mocks/shapes.json#L28-L32

The first 2 have a log y axis, and the values are as data. The next two have date x axis, and the values are date strings (with an underscore between date and time for `path` shapes)

But it _is_ true for annotations on `log` axes only that you need to take the log of the desired position. See `annotations_autorange.json` where `y: 2` anchors to data value 100:

https://github.com/plotly/plotly.js/blob/b194d8c3ef503ac1989666547cccb65dc60fb44f/test/image/mocks/annotations-autorange.json#L72-L76